drv/stm32xx-i2c/src/lib.rs

```diff
@@ -641,7 +556,7 @@ impl I2cController<'_> {
                         break;
                     }
 
-                    self.wfi()?;
+                    self.wfi();
                     sys_irq_control(notification, true);
```

Comment:
Perhaps better suited for a follow-up PR, but it looks like all the callers of `wfi` immediately call `sys_irq_control` with the notification mask that was waited on...what do you think about having `wfi` just do that?